flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java

```diff
@@ -390,10 +390,16 @@ public InputDependencyConstraint getInputDependencyConstraint() {
 					jobVertex.getInvokableClassName(),
 					jobVertex.getConfiguration());
 
-				taskInformationOrBlobKey = BlobWriter.serializeAndTryOffload(
-					taskInformation,
-					getJobId(),
-					blobWriter);
+				try {
+					taskInformationOrBlobKey = BlobWriter.serializeAndTryOffload(
+						taskInformation,
+						getJobId(),
+						blobWriter);
+				} catch (IOException e) {
+					throw new ExecutionGraphException(
```

Comment:
Why we need this exception wrapper?

Reply:
True, just some leftover from another refactoring attempt.